raiden-ts/src/path/epics.ts

```diff
@@ -111,8 +217,13 @@ export const pathFindServiceEpic = (
                       pluck(0),
                     );
                 return pfs$.pipe(
-                  // TODO: use pfs.price to create & sign IOU
                   mergeMap(pfs =>
+                    pfs.price.isZero()
+                      ? of(undefined)
+                      : prepareNextIOU$(state, pfs, deps, httpTimeout, tokenNetwork),
+                  ),
+                  withLatestFrom(pfs$),
```

Comment:
Don't reuse it. Notice `pfs$` can be a cold observable which performs some [slow] network request. Instead, either return an explicit `of(pfs)` or put the body of `mergeMap` below also under/inside this mergeMap above, of `prepareNextIOU$`. There're tons of example of this pattern around, tell me if you have any question.